### PR TITLE
refactor(protocol): use more abi.encodePacked

### PR DESCRIPTION
**Pull request by dantaik**

### CODE REVIEW

File: packages/protocol/test/L1/TaikoL1TestBase.sol [hunk 1]
```diff
@@ -328,7 +328,7 @@ abstract contract TaikoL1TestBase is TestBase {
         returns (bytes memory signature)
     {
         bytes32 digest = keccak256(
-            abi.encode(
```
Comment (dantaik) [inline packages/protocol/test/L1/TaikoL1TestBase.sol:331]: better to use `getSignedHash` from SgxVerifier.sol without rewriting the code

File: packages/protocol/contracts/tokenvault/ERC20Vault.sol [hunk 2]
```diff
@@ -348,7 +348,7 @@ contract ERC20Vault is
             owner(),
             bytes.concat(
                 ProxiedBridgedERC20(bridgedToken).init.selector,
-                abi.encode(
+                abi.encodePacked(
```
Comment (adaki2004) [inline packages/protocol/contracts/tokenvault/ERC20Vault.sol:351]: Same applies as https://github.com/taikoxyz/taiko-mono/pull/15023/files#r1369851786

File: packages/protocol/contracts/tokenvault/ERC721Vault.sol [hunk 3]
```diff
@@ -299,7 +299,7 @@ contract ERC721Vault is
             owner(),
             bytes.concat(
                 ProxiedBridgedERC721(bridgedToken).init.selector,
-                abi.encode(
+                abi.encodePacked(
```
Comment (adaki2004) [inline packages/protocol/contracts/tokenvault/ERC721Vault.sol:302]: Same as: https://github.com/taikoxyz/taiko-mono/pull/15023/files#r1369851786

File: packages/protocol/contracts/tokenvault/ERC1155Vault.sol [hunk 4]
```diff
@@ -352,7 +352,7 @@ contract ERC1155Vault is BaseNFTVault, ERC1155ReceiverUpgradeable {
             owner(),
             bytes.concat(
                 ProxiedBridgedERC1155(bridgedToken).init.selector,
-                abi.encode(
+                abi.encodePacked(
```
[the remark this answers is not in the document]
Comment (dantaik) [reply packages/protocol/contracts/tokenvault/ERC1155Vault.sol:355]: Good to know that. Reverting.

File: packages/protocol/test/L1/TaikoToken.t.sol [hunk 5]
```diff
@@ -31,7 +31,7 @@ contract TaikoTokenTest is TestBase {
             address(tko),
             bytes.concat(
                 tko.init.selector,
-                abi.encode(
+                abi.encodePacked(
```
Comment (adaki2004) [inline packages/protocol/test/L1/TaikoToken.t.sol:34]: Same as the resolved one - we need `.encode ` here.

**Pull request merged**